File: C#/spavanac.cs
using System;

public class spavanac {
    public static void Main (string[] args) {
        string line = Console.ReadLine ();
        int space = line.IndexOf (' ');
        int hour = int.Parse (line.Substring (0, space));
        int min = int.Parse (line.Substring (space + 1)) - 45;
        if (min < 0) {
            min += 60;
            hour--;
            if (hour < 0)
                hour = 23;
        }
        Console.WriteLine (hour + " " + min);
    }
}

File: Java/spavanac.java

//  https://open.kattis.com/problems/spavanac
import java.util.Scanner;

public class spavanac {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int hour = sc.nextInt();
        int min = sc.nextInt() - 45;
        if (min < 0) {
            min += 60;
            hour--;
            if (hour < 0)
                hour = 23;
        }
        System.out.println(hour + " " + min);
        sc.close();
    }
}